target/linux/ath79/nand/base-files/lib/upgrade/platform.sh

```diff
@@ -7,13 +7,27 @@ platform_check_image() {
 	return 0
 }
 
-RAMFS_COPY_BIN='fw_printenv fw_setenv'
+RAMFS_COPY_BIN='fw_printenv fw_setenv nandwrite'
 RAMFS_COPY_DATA='/etc/fw_env.config /var/lock/fw_printenv.lock'
 
+platform_do_upgrade_mikrotik_rb() {
+	CI_KERNPART=none
+	local fw_mtd=$(find_mtd_part kernel)
+	fw_mtd="${fw_mtd/block/}"
+	[ -n "$fw_mtd" ] || return
+	mtd erase kernel
+	tar xf "$1" sysupgrade-routerboard/kernel -O | nandwrite -o "$fw_mtd" -
+
+	nand_do_upgrade "$1"
+}
+
 platform_do_upgrade() {
 	local board=$(board_name)
 
 	case "$board" in
+	mikrotik,routerboard-922uags-5hpacd)
+		platform_do_upgrade_mikrotik_rb "$1"
```

Comment:
Consider renaming that based on the pattern used for glinet below:
mikrotik_rb_do_upgrade

I do not know what's common elsewhere though, so might be the opposite as well and your style is better ...

Reply:
As I said above, I just got it as it was in ar71xx, although this does not mean it's good _per se_. Other targets/devices use:

- platform_do_upgrade_linksys (mvebu, kirkwood, ipq40xx)
- platform_do_upgrade_openmesh (ipq40xx)
- platform_do_upgrade_seama (brcm53xx)
- zyxel_do_upgrade (ipq40xx)
- mbl_do_upgrade (apm821xx)
- glinet_nand_nor_do_upgrade (ath79)

So there doesn't seem to be a consistent way to do it. In any case, I'll replace "mikrotik_rb" by "mikrotik_nand".

Reply:
Okay, keep it as you wish.